airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/manifest.yaml

```diff
@@ -1167,6 +1183,159 @@ definitions:
+          - "%Y-%m-%d"
+        datetime_format: "%Y-%m-%d"
+        lookback_window: P{{ config.get('look_back_window', 0) }}D
+        start_datetime:
+          type: MinMaxDatetime
+          datetime: "{{ config.get('start_date', today_utc().strftime('%Y-%m-%d') ) }}"
+          datetime_format: "%Y-%m-%d"
+        end_datetime:
+          datetime: "{{ today_utc().strftime('%Y-%m-%d') }}"
+          datetime_format: "%Y-%m-%d"
+        cursor_granularity: "P1D"
+        step: "P{{ config.get('events_time_step', 30) }}D"
+        start_time_option:
+          field_name: startDate
+          inject_into: body_json
+        end_time_option:
+          field_name: endDate
+          inject_into: body_json
+      transformations:
+        - type: AddFields
+          fields:
+            - path: ["reportDate"]
+              value: "{{ stream_slice.end_time }}"
+        - type: AddFields
+          fields:
+            - path: ["profileId"]
+              value: "{{ stream_slice['profileId'] }}"
+
+      retriever:
+        type: AsyncRetriever
+        partition_router:
+          type: SubstreamPartitionRouter
+          parent_stream_configs:
+            - type: ParentStreamConfig
+              parent_key: "profileId"
+              stream: "#/definitions/streams/profiles_filtered"
+              partition_field: "profileId"
+        status_mapping:
+          failed:
+            - FAILED
+          running:
+            - PENDING
+            - PROCESSING
+          timeout:
+            - timeout
+          completed:
+            - COMPLETED
+        urls_extractor:
+          type: DpathExtractor
+          field_path:
+            - url
+        record_selector:
+          type: RecordSelector
+          extractor:
+            type: DpathExtractor
+            field_path: []
+
+        status_extractor:
+          type: DpathExtractor
+          field_path:
+            - status
+        creation_requester:
+          type: HttpRequester
+          path: /reporting/reports
+          url_base: >-
+            {%- set URL_MAPPING = {
+                "NA": "https://advertising-api.amazon.com/",
+                "EU": "https://advertising-api-eu.amazon.com/",
+                "FE": "https://advertising-api-fe.amazon.com/"
+            } -%}
+            {{ URL_MAPPING[config["region"]] }}
+          authenticator: "#/definitions/oauth_authenticator"
+          http_method: POST
+          request_headers:
+            Amazon-Advertising-API-Scope: "{{ stream_slice['profileId'] }}"
+            Amazon-Advertising-API-ClientId: '{{ config["client_id"] }}'
+          request_body_json:
+            name: "purchasedAsin report {{ stream_slice.start_time }}"
+            startDate: "{{ stream_slice.start_time }}"
+            endDate: "{{ stream_slice.end_time }}"
+            configuration:
+              adProduct: SPONSORED_BRANDS
+              groupBy: ["purchasedAsin"]
+              columns: # TODO: make configurable
```

Comment:
is this relevant? i think we have all the fields covered and doesn't look like this stream supports custom metrics

Reply:
yes; I've added a task in https://github.com/airbytehq/airbyte-internal-issues/issues/10429 :
> https://github.com/airbytehq/airbyte-internal-issues/issues/10429

this stream does not support custom metrics; but we can remove some of them and therefore exclude them from request